light-client-verifier/src/operations/commit_validator.rs

```diff
@@ -11,6 +14,9 @@ use crate::{
 /// Validates the commit associated with a header against a validator set
 pub trait CommitValidator: Send + Sync {
     /// Perform basic validation
+    #[cfg(feature = "rust-crypto")]
+    type RustCryptoProvider: CryptoProvider;
+
```

Comment:
Feature-gating trait members is not a good idea, I think.

Perhaps I did not explain the idea more carefully. It's OK to work the abstract `CryptoProvider` generics wherever needed without feature-gating. So you could have (with tentative naming):

```rust
// Kill the CommitValidator trait, the methods become inherent members of the struct:

pub struct CommitValidator<C> {
    // ...
}

impl<C: CryptoProvider> CommitValidator<C> {
    // The API that belonged to CommitValidator trait goes here
}

/// The batteries-included validator, for when you don't mind the dependencies on
/// the full rust-crypto stack.
#[cfg(feature = "rust-crypto")]
pub type ProdCommitValidator = CommitValidator<RustCryptoProvider>;
```

@thanethomson I think all those traits having a single empty `ProdFoo` implementation don't carry much weight in the current design, since all the logic is provided in the trait methods. So it's fair to replace them with generic types parameterized around the crypto provider, like in the example above.